modules/migration/migration-5.0.0_to_5.1.0/wso2-is-migration-client/src/main/java/org/wso2/carbon/is/migration/client/MigrateFrom5to510.java

```diff
@@ -452,31 +453,50 @@ public void migrateIdentityDBFinalize(){
                 alterTokenIdNotNull = SQLQueries.ALTER_TOKEN_ID_NOT_NULL_MYSQL;
             }
             String setAccessTokenPrimaryKey = SQLQueries.SET_ACCESS_TOKEN_PRIMARY_KEY;
+
+            String alterCodeIdNotNull;
+            if ("oracle".equals(databaseType)){
+                alterCodeIdNotNull = SQLQueries.ALTER_CODE_ID_NOT_NULL_ORACLE;
+            } else if ("mssql".equals(databaseType)){
+                alterCodeIdNotNull = SQLQueries.ALTER_CODE_ID_NOT_NULL_MSSQL;
+            } else if ("postgresql".equals(databaseType)){
+                alterCodeIdNotNull = SQLQueries.ALTER_CODE_ID_NOT_NULL_POSTGRESQL;
+            } else if ("h2".equals(databaseType)) {
+                alterCodeIdNotNull = SQLQueries.ALTER_CODE_ID_NOT_NULL_H2;
+            } else {
+                alterCodeIdNotNull = SQLQueries.ALTER_CODE_ID_NOT_NULL_MYSQL;
+            }
             String setAuthorizationCodePrimaryKey = SQLQueries.SET_AUTHORIZATION_CODE_PRIMARY_KEY;
             String setScopeAssociationPrimaryKey = SQLQueries.SET_SCOPE_ASSOCIATION_PRIMARY_KEY;
 
             dropColumnPS = identityConnection.prepareStatement(dropTokenScopeColumn);
             dropColumnPS.execute();
 
-            notNullPS = identityConnection.prepareStatement(alterTokenIdNotNull);
-            notNullPS.execute();
+            tokenIdNotNullPS = identityConnection.prepareStatement(alterTokenIdNotNull);
+            tokenIdNotNullPS.execute();
 
             primaryKeyPS = identityConnection.prepareStatement(setAccessTokenPrimaryKey);
             primaryKeyPS.execute();
 
+            codeIdNotNullPS = identityConnection.prepareStatement(alterCodeIdNotNull);
```

Comment:
In our test DB there are still 6k records with code_id = null (we have migrated it w/o this constraint)
So I assume the fix will fail again when perform this alter
Probably you should delete/update data in IDN_OAUTH2_AUTHORIZATION_CODE table before switching the constraint on.